### PR TITLE
Feat: Apache Kafka using Strimzi on EKS

### PR DESCRIPTION
**Pull request by AlyIbrahim**
### What does this PR do?

🛑 Please open an issue first to discuss any significant work and flesh out details/direction - we would hate for your time to be wasted.
Consult the [CONTRIBUTING](https://github.com/awslabs/data-on-eks/blob/main/CONTRIBUTING.md#contributing-via-pull-requests) guide for submitting pull-requests.

This PR has templates to create Kafka on EKS using Strimzi with metrics System that uses Prometheus and Grafana dashboards, provided by Strimzi, configured on the EKS.

### Motivation

https://github.com/awslabs/data-on-eks/issues/22

### More

- [x] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [x] Yes, I have added a new example under [examples](https://github.com/awslabs/data-on-eks/tree/main/examples) to support my PR
- [x] Yes, I have created another PR to add a new Add-on for [Terraform EKS Blueprints](https://github.com/aws-ia/terraform-aws-eks-blueprints) repo (if applicable)
- [x] Yes, I have updated the `website/docs` or `website/blog` section for this feature
- [x] Yes, I ran `pre-commit run -a` with this PR. Link for installing [pre-commit](https://pre-commit.com/) locally

### For Moderators

- [x] E2E Test successfully complete before merge?

### Additional Notes

This PR has dependency on PR https://github.com/aws-ia/terraform-aws-eks-blueprints/pull/1090
Updates to READ ME will follow

### CODE REVIEW

**Comment by nabuskey** [comment]
Please mark this as draft if not ready for review.

**Comment by AlyIbrahim** [comment]
PR https://github.com/aws-ia/terraform-aws-eks-blueprints/pull/1090 has been merged and now this PR can be tested

**Review by nabuskey — COMMENTED** [review]
Thank you very much for this! 
Please add readme with instructions and what this does, currently the README file is empty and users won't be able to tell what to do with this. 
I think it's a good idea to have the same directory structure for everything we have in this repo. 
Also jobs are still failing. Looks like it's still referencing modules don't have what you are expecting.

File: streaming/kafka/kubernetes-manifests/gp3-storage-class.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+apiVersion: storage.k8s.io/v1
```
[the remark this answers is not in the document]
Comment (AlyIbrahim) [reply streaming/kafka/kubernetes-manifests/gp3-storage-class.yml]: Yep, this is redundant. Removed.

File: streaming/kafka/kubernetes-manifests/grafana-manifests/grafana.yml [hunk 2]
```diff
@@ -0,0 +1,65 @@
+# This is not a recommended configuration, and further support should be available
```
[the remark this answers is not in the document]
Comment (AlyIbrahim) [reply streaming/kafka/kubernetes-manifests/grafana-manifests/grafana.yml]: This is redundant. Removed.

File: streaming/kafka/values.yaml [hunk 3]
```diff
@@ -0,0 +1,28 @@
+# Default values for strimzi-kafka-operator.
```
[the remark this answers is not in the document]
Comment (AlyIbrahim) [reply streaming/kafka/values.yaml]: Moved to helm-values dir to match the directory structure.

**Comment by AlyIbrahim** [comment]
Added documentation and README, and modified the directory structure.

File: streaming/kafka/addons.tf [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+  enable_prometheus                   = true
+  # enable_grafana                      = true
+
+  # Kafka (Strimzi) add-on with custom helm config
+  enable_kafka = true
+  kafka_helm_config = {
+    name             = local.strimzi_kafka_name
+    chart            = local.strimzi_kafka_name
+    repository       = "https://strimzi.io/charts/"
+    version          = "0.31.1"
```
[the remark this answers is not in the document]
Comment (vara-bonthu) [reply streaming/kafka/addons.tf:158]: We hardcode this all the helm charts in all our patterns.

File: streaming/kafka/values.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (vara-bonthu) [reply streaming/kafka/values.yaml]: fixed

**Review by vara-bonthu — APPROVED** [review]
Great PR @AlyIbrahim 👍🏼 
I have pushed few changes to the PR and tested. All looks good.
I have also updated the doc with more details and screenshots from Grafana etc.

**Comment by nabuskey** [comment]
LGTM. Thank you very much for this!
@vara-bonthu We may want to move website related stuff to another PR in the future. I think PRs for new services are large enough. I think it's better to cut down the amount of things we have to look through in a PR.

**Pull request merged**